spec/robots/dor_repo/accession/end_accession_spec.rb

```diff
@@ -27,7 +27,7 @@
 
     context 'when there is no special dissemniation workflow' do
       it 'cleans up' do
-        perform
+        expect(perform.status).to eq 'noop'
```

Comment:
or does "perform.status" first do the perform and then return the status?   because this is a sort of method name chain?

Reply:
yeah, i think that's what's happening.  i believe the "status" here is the lybercore "noop" code indicating that the robot did none of the actual desired work, but delegated it to another service, which will then mark the workflow step complete once the work's actually been performed.  otherwise, by default, once `perform`/`perform_work` returns, the workflow step would be marked complete (via some lybercore magic?), when it's more likely that DSA hasn't actually done the work yet.

confusing because it looks similar to `ActiveJob`'s `perform` and `perform_later` methods, the latter of which wouldn't give you any status info other than whether the job had been queued or not.

i think the goal here is to do something similar to what we do in the pres bots `validate-moab` handoff to a pres cat worker, or the [common-accessioning handoff to DSA](https://github.com/sul-dlss/common-accessioning/pull/1081) to do the actual work for the `update-marc` step.  both of those use this `noop` pattern.

Reply:
Refactored to make this a tad clearer.